nuxeo-ai-core/src/main/java/org/nuxeo/ai/services/PersistedConfigurationServiceImpl.java

```diff
@@ -104,22 +124,41 @@ public Descriptor retrieve(String key) throws IOException {
 
     @Override
     public String toXML(String tag, List<? extends Descriptor> descriptors) throws IOException {
-        StringBuilder builder = new StringBuilder("<?xml version=\"1.0\"?>");
-        builder.append("<");
-        builder.append(tag);
-        builder.append(">");
-
+        String empty = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><" + tag + "/>";
+        if (descriptors.isEmpty())
+            return empty;
+        // create root element
         XMap xmap = reader.getXMap();
-        for (Descriptor descriptor : descriptors) {
-            String xml = xmap.toXML(descriptor);
-            builder.append(xml);
+        DocumentBuilder docBuilder;
+        Element root;
+        try {
+            docBuilder = dbfac.newDocumentBuilder();
+            Document doc = docBuilder.newDocument();
+            root = doc.createElement(tag);
+            for (Descriptor descriptor : descriptors) {
+                xmap.toXML(descriptor, root);
+            }
```

Comment:
1. It would be nice if you can explain your logic here. 
You're trying to map all given descriptors to an XML element. But there is no verification that they are instances of the same class. It's an issue even in the old code - true. 

2. I guess, the advantage of using `Document` is to have an exception. 
3. `XMap#toXML(Object, Element)` is not clear on its intentions in terms of if it appends new tags or simply replaces

Reply:
1. The check is already done at the upper level

2. I dont get the exception part

3. Its adding children to the document `root`

The whole reason of this new code is to avoid having such contributions:

```
<xml....>
<root>
<thresolds>
    <threshold id="first"/>
<thresholds/>
</root>
<root>
<thresholds>
   <threshold id="second"/>
</thresholds>
</root>
```

The `root` is added in an hardcoded way of `XMAP#toXML(Object)` which is a bit toxic.